src/main/java/org/disq_bio/disq/impl/formats/bam/BamRecordGuesser.java

```diff
@@ -170,8 +170,11 @@ private Result checkRecordStartInternal(long vPos) throws IOException {
     int cigarOpsLength = numCigarOps * 4;
     int seqLength = buf.getInt(20) + (buf.getInt(20) + 1) / 2;
 
-    if ((flags & 4) == 0 && (seqLength == 0 || numCigarOps == 0)) {
-      return NO_START; // Non-empty cigar/seq in mapped reads
+    boolean mapped = (flags & 4) == 0;
+    boolean secondary = (flags & 256) != 0;
+    if (mapped && ((seqLength == 0 && !secondary) || numCigarOps == 0)) {
```

Comment:
Could you add a test demonstrating this behavior?

Reply:
Done. The test fails without the change.